features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Shared/ApiKeyRestriction.jsx

```diff
@@ -0,0 +1,340 @@
+        <Typography variant="h5" component="h6">
+          Key Restrictions
+        </Typography>
+        <FormControl component="fieldset">
+          <RadioGroup
+            aria-label="API Key Restrictions"
+            value={restrictSchema}
+            row
+            onChange={onRestrictSchemaChange}
+          >
+            <FormControlLabel
+              value="none"
+              control={<Radio color="primary" />}
+              label="None"
+              labelPlacement="end"
+            />
+            <FormControlLabel
+              value="ip"
+              control={<Radio color="primary" />}
+              label="IP Addresses"
+              labelPlacement="end"
+            />
+            <FormControlLabel
+              value="referer"
+              control={<Radio color="primary" />}
+              label="HTTP Referrers (Web Sites)"
+              labelPlacement="end"
+            />
+          </RadioGroup>
+        </FormControl>
+
+        {restrictSchema === "ip" && (
+          <Box component="div" id="ipPanel">
+            <Grid
+              container
+              direction="row"
+              spacing={0}
+              justify="left"
+              alignItems="left"
+            >
+              <Grid item md={10} xs={10}>
+                <TextField
+                  label="IP Address"
+                  value={newIP}
+                  onChange={onIpTextUpdate}
+                  className={classes.inputText}
+                  helperText={
+                    invalidIP
+                      ? intl.formatMessage({
+                          defaultMessage: "Invalid IP Address",
+                          id:
+                            "Shared.AppsAndKeys.Tokens.apiKeyRestriction.ip.validity.error",
+                        })
+                      : ""
+                  }
+                  error={invalidIP}
+                  margin="dense"
+                  variant="outlined"
+                  placeholder={intl.formatMessage({
+                    defaultMessage: "Enter IP Address",
+                    id: "Shared.AppsAndKeys.Tokens.apiKeyRestriction.enter.ip",
+                  })}
+                  fullWidth
+                />
+              </Grid>
+              <Grid item md={2} xs={2}>
+                <span>
+                  <Fab
+                    className={classes.Fab}
+                    size="small"
+                    color="primary"
+                    aria-label="add"
+                    onClick={addIpItem}
+                  >
+                    <AddIcon />
+                  </Fab>
+                </span>
+              </Grid>
+            </Grid>
+            <Grid
+              container
+              direction="row"
+              spacing={0}
+              justify="left"
+              alignItems="left"
+              md={10}
+              xs={10}
+            >
+              {ipList.length > 0 && (
```

Comment:
need to have a space before open curly and after close curly. Fix elsewhere as well.

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/9250